analytics/lib/counts.py

```diff
@@ -258,14 +258,18 @@ def do_pull_from_zerver(stat, start_time, end_time, interval):
         INSERT INTO analytics_realmcount
             (realm_id, value, property, end_time, interval)
     SELECT
-        zerver_stream.realm_id, count(*), '%(property)s', %%(time_end)s, '%(interval)s'
-    FROM zerver_stream
-    LEFT JOIN zerver_recipient
+        zerver_realm.id, count(*), '%(property)s', %%(time_end)s, '%(interval)s'
+    FROM zerver_realm
+    LEFT JOIN zerver_stream
     ON
     (
-        zerver_recipient.type = 2 AND
-        zerver_stream.id = zerver_recipient.type_id
+        zerver_stream.realm_id = zerver_realm.id AND
+        zerver_stream.date_created >= %%(time_start)s AND
+        zerver_stream.date_created < %%(time_end)s
+        %(join_args)s
     )
+    WHERE
+        zerver_realm.date_created < %%(time_end)s
     GROUP BY zerver_stream.realm_id
```

Comment:
I'd do `GROUP BY zerver_realm.id` for symmetry with the other count_X_by_Y_query strings.